pkg/devfile/adapters/kubernetes/component/push.go

```diff
@@ -34,6 +40,46 @@ func (a *Adapter) getComponentDeployment() (*appsv1.Deployment, bool, error) {
 	return deployment, componentExists, nil
 }
 
+func (a *Adapter) buildPushAutoImageComponents(ctx context.Context, fs filesystem.Filesystem, devfileObj parser.DevfileObj, compStatus *watch.ComponentStatus) error {
+	components, err := libdevfile.GetImageComponentsToPush(devfileObj)
+	if err != nil {
+		return err
+	}
+
+	for _, c := range components {
+		if c.Image == nil {
+			return fmt.Errorf("component %q should be an Image Component", c.Name)
+		}
+		alreadyApplied, ok := compStatus.ImageComponentsAutoApplied[c.Name]
+		if ok && reflect.DeepEqual(*c.Image, alreadyApplied) {
+			klog.V(3).Infof("Skipping image component %q; already applied and not changed", c.Name)
+			continue
+		}
+		err = image.BuildPushSpecificImage(ctx, fs, c, true)
+		if err != nil {
+			return err
+		}
+		compStatus.ImageComponentsAutoApplied[c.Name] = *c.Image
+	}
+
+	// Remove keys that might no longer be valid
+	devfileHasCompFn := func(n string) bool {
+		for _, c := range components {
+			if c.Name == n {
```

Comment:
Would this account for any changes that might have happened to the component? I assume it is already accounted for at this point owing to [L#54](https://github.com/redhat-developer/odo/pull/6654/files#diff-014b2433b46bcae145f844d9d2e8560550acf129093f2691bc8be2097eac1a0aR54), but I'd still like to confirm.

Reply:
Yes, changes to the component itself are taken into account with the deep comparison in [L54](https://github.com/redhat-developer/odo/pull/6654/files#diff-014b2433b46bcae145f844d9d2e8560550acf129093f2691bc8be2097eac1a0aR54).
This `devfileHasCompFn` aims at just cleaning up items from the cache to account for eventual cases where the component is removed from the Devfile.